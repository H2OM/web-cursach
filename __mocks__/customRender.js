import ClientContext from '@/lib/context/ClientContext';
import { render } from '@testing-library/react';

export default customRender = (ui, {providerProps}) => {
    return render(
      <ClientContext.Provider value={{...providerProps}}>{ui}</ClientContext.Provider>,
    )
}