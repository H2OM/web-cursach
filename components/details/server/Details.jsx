import Uptitle from "@/lib/basecomponents/uptitle/uptitle";
import Image from "next/image";
import '../details.scss';
import Ratingbar from "@/lib/basecomponents/raitingbar/ratingbar";
import { notFound } from "next/navigation";
import Addtofav from "@/lib/basecomponents/addtofav/addtofav";

export default async function Details ({params}) {
    const data = await fetch('http://127.0.0.1/api/catalog/cart-detail?article='+params.details, {method: 'GET', cache: "no-cache"})
    .then(data=>{
        if(!data.ok) {
            return false;

        } 
        return data.json();
    }).catch(()=>false);
    

    if(!data || typeof(data) !== 'object' || data.title === undefined) {
        return notFound();
    }

    return (
        <section className="Details">
            <div className="container">
                <Uptitle full><h1 className="title">{data.title}</h1></Uptitle>
                
                <div className="Details__split">
                    <Ratingbar rating={data.rating} voices={data.voices} userrating={data.userrating} id={data.id} id2={data.image}/>
                    <div className="Details__split__p"><Addtofav id={data.id} title={data.title} article={data.article} image={data.image}/>{data.city}</div>
                </div>
                <Image
                    src={"/img/"+data.image}
                    alt="image"
                    className="Details__image"
                    width={0}
                    height={0}
                    sizes='100vw'
                    priority={true}
                    quality={100}
                />
                <div className="Details__text">{data.text}</div>
            </div>
        </section>
    )
}