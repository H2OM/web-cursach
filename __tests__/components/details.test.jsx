import settingFetchResult from "@/__mocks__/fetchMock";
import 'intersection-observer';
import { usePathname, useSearchParams, useRouter } from 'next/navigation';
import customRender from "@/__mocks__/customRender";
import { screen } from "@testing-library/react";
import Details from "@/components/details/server/Details";

jest.mock('next/navigation');
useRouter.mockReturnValue({
    push: jest.fn(),
}); 
usePathname.mockReturnValue("localhost");
useSearchParams.mockReturnValue({});

it('details test',async ()=>{
    const data = {
        article: 'OZERO',
        image: '/img/image1.jpg',
        title: 'ozero',
        id: 3,
        text: 'sometext',
        rating: 5,
        voices: 3
    };
    settingFetchResult(data);

    const {container} = customRender(await Details({params: {details: "OZERO"}}), {providerProps: {favorites: [3,4]}});

    expect(await screen.findByText('ozero')).toBeInTheDocument();
    expect(screen.getByRole('img')).toBeInTheDocument();
    expect(await screen.getByText('sometext')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
});


