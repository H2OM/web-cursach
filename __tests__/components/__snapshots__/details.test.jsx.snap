// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`details test 1`] = `
<div>
  <section
    class="Details"
  >
    <div
      class="container"
    >
      <div
        class="Uptitle__wrap"
      >
        <div
          class="Uptitle__anchorUpUptitle__anchorUp_full "
        />
        <div
          class="Uptitle Uptitle_full "
        >
          <h1
            class="title"
          >
            ozero
          </h1>
        </div>
        <div
          class="Uptitle__anchorDownUptitle__anchorDown_full "
        />
      </div>
      <div
        class="Details__split"
      >
        <div
          class="Ratingbar "
        >
          <div
            class="Ratingbar__rait"
          >
            <svg
              class="Ratingbar__rait__star "
              fill="none"
              height="24"
              role="svg"
              viewBox="0 0 26 24"
              width="26"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.88061 23.07L5.88056 23.0702C5.85306 23.2264 5.91485 23.3546 6.00976 23.4299C6.1027 23.5036 6.21513 23.5201 6.32514 23.4639L5.88061 23.07ZM5.88061 23.07L7.17748 15.6793L7.22376 15.4156L7.02987 15.2309L1.52553 9.98747C1.52549 9.98743 1.52544 9.98739 1.5254 9.98734C1.41972 9.88639 1.38802 9.74107 1.42943 9.60208C1.47062 9.46381 1.56872 9.37728 1.69212 9.35993L1.69285 9.35982L9.34597 8.27232L9.60439 8.2356L9.72199 8.00259L13.1344 1.24179C13.2573 0.998504 13.5674 0.998504 13.6903 1.24179L17.1027 8.00259L17.2203 8.2356L17.4787 8.27232L25.1318 9.35982L25.1326 9.35993C25.256 9.37728 25.354 9.4638 25.395 9.60179C25.4362 9.74044 25.4045 9.88573 25.2982 9.98695L25.2979 9.98719L19.7948 15.2309L19.6009 15.4156L19.6472 15.6793L20.9441 23.07L20.9441 23.0702C20.9716 23.2264 20.9099 23.3546 20.8149 23.4299C20.7219 23.5036 20.6094 23.5201 20.4993 23.4637C20.4993 23.4637 20.4992 23.4637 20.4991 23.4637L13.6385 19.9388L13.4099 19.8214L13.1814 19.9389L6.32555 23.4637L5.88061 23.07Z"
                fill="url(#paint3/img/image1.jpg0_linear_73_120)"
                stroke="white"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="paint3/img/image1.jpg0_linear_73_120"
                  x1="0.91211"
                  x2="25.5049"
                  y1="11.7199"
                  y2="11.7199"
                >
                  <stop
                    stop-color="white"
                  />
                  <stop
                    offset="1"
                    stop-color="white"
                  />
                  <stop
                    offset="1"
                    stop-color="white"
                    stop-opacity="0"
                  />
                </lineargradient>
              </defs>
            </svg>
            <svg
              class="Ratingbar__rait__star "
              fill="none"
              height="24"
              role="svg"
              viewBox="0 0 26 24"
              width="26"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.88061 23.07L5.88056 23.0702C5.85306 23.2264 5.91485 23.3546 6.00976 23.4299C6.1027 23.5036 6.21513 23.5201 6.32514 23.4639L5.88061 23.07ZM5.88061 23.07L7.17748 15.6793L7.22376 15.4156L7.02987 15.2309L1.52553 9.98747C1.52549 9.98743 1.52544 9.98739 1.5254 9.98734C1.41972 9.88639 1.38802 9.74107 1.42943 9.60208C1.47062 9.46381 1.56872 9.37728 1.69212 9.35993L1.69285 9.35982L9.34597 8.27232L9.60439 8.2356L9.72199 8.00259L13.1344 1.24179C13.2573 0.998504 13.5674 0.998504 13.6903 1.24179L17.1027 8.00259L17.2203 8.2356L17.4787 8.27232L25.1318 9.35982L25.1326 9.35993C25.256 9.37728 25.354 9.4638 25.395 9.60179C25.4362 9.74044 25.4045 9.88573 25.2982 9.98695L25.2979 9.98719L19.7948 15.2309L19.6009 15.4156L19.6472 15.6793L20.9441 23.07L20.9441 23.0702C20.9716 23.2264 20.9099 23.3546 20.8149 23.4299C20.7219 23.5036 20.6094 23.5201 20.4993 23.4637C20.4993 23.4637 20.4992 23.4637 20.4991 23.4637L13.6385 19.9388L13.4099 19.8214L13.1814 19.9389L6.32555 23.4637L5.88061 23.07Z"
                fill="url(#paint3/img/image1.jpg1_linear_73_120)"
                stroke="white"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="paint3/img/image1.jpg1_linear_73_120"
                  x1="0.91211"
                  x2="25.5049"
                  y1="11.7199"
                  y2="11.7199"
                >
                  <stop
                    stop-color="white"
                  />
                  <stop
                    offset="1"
                    stop-color="white"
                  />
                  <stop
                    offset="1"
                    stop-color="white"
                    stop-opacity="0"
                  />
                </lineargradient>
              </defs>
            </svg>
            <svg
              class="Ratingbar__rait__star "
              fill="none"
              height="24"
              role="svg"
              viewBox="0 0 26 24"
              width="26"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.88061 23.07L5.88056 23.0702C5.85306 23.2264 5.91485 23.3546 6.00976 23.4299C6.1027 23.5036 6.21513 23.5201 6.32514 23.4639L5.88061 23.07ZM5.88061 23.07L7.17748 15.6793L7.22376 15.4156L7.02987 15.2309L1.52553 9.98747C1.52549 9.98743 1.52544 9.98739 1.5254 9.98734C1.41972 9.88639 1.38802 9.74107 1.42943 9.60208C1.47062 9.46381 1.56872 9.37728 1.69212 9.35993L1.69285 9.35982L9.34597 8.27232L9.60439 8.2356L9.72199 8.00259L13.1344 1.24179C13.2573 0.998504 13.5674 0.998504 13.6903 1.24179L17.1027 8.00259L17.2203 8.2356L17.4787 8.27232L25.1318 9.35982L25.1326 9.35993C25.256 9.37728 25.354 9.4638 25.395 9.60179C25.4362 9.74044 25.4045 9.88573 25.2982 9.98695L25.2979 9.98719L19.7948 15.2309L19.6009 15.4156L19.6472 15.6793L20.9441 23.07L20.9441 23.0702C20.9716 23.2264 20.9099 23.3546 20.8149 23.4299C20.7219 23.5036 20.6094 23.5201 20.4993 23.4637C20.4993 23.4637 20.4992 23.4637 20.4991 23.4637L13.6385 19.9388L13.4099 19.8214L13.1814 19.9389L6.32555 23.4637L5.88061 23.07Z"
                fill="url(#paint3/img/image1.jpg2_linear_73_120)"
                stroke="white"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="paint3/img/image1.jpg2_linear_73_120"
                  x1="0.91211"
                  x2="25.5049"
                  y1="11.7199"
                  y2="11.7199"
                >
                  <stop
                    stop-color="white"
                  />
                  <stop
                    offset="1"
                    stop-color="white"
                  />
                  <stop
                    offset="1"
                    stop-color="white"
                    stop-opacity="0"
                  />
                </lineargradient>
              </defs>
            </svg>
            <svg
              class="Ratingbar__rait__star "
              fill="none"
              height="24"
              role="svg"
              viewBox="0 0 26 24"
              width="26"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.88061 23.07L5.88056 23.0702C5.85306 23.2264 5.91485 23.3546 6.00976 23.4299C6.1027 23.5036 6.21513 23.5201 6.32514 23.4639L5.88061 23.07ZM5.88061 23.07L7.17748 15.6793L7.22376 15.4156L7.02987 15.2309L1.52553 9.98747C1.52549 9.98743 1.52544 9.98739 1.5254 9.98734C1.41972 9.88639 1.38802 9.74107 1.42943 9.60208C1.47062 9.46381 1.56872 9.37728 1.69212 9.35993L1.69285 9.35982L9.34597 8.27232L9.60439 8.2356L9.72199 8.00259L13.1344 1.24179C13.2573 0.998504 13.5674 0.998504 13.6903 1.24179L17.1027 8.00259L17.2203 8.2356L17.4787 8.27232L25.1318 9.35982L25.1326 9.35993C25.256 9.37728 25.354 9.4638 25.395 9.60179C25.4362 9.74044 25.4045 9.88573 25.2982 9.98695L25.2979 9.98719L19.7948 15.2309L19.6009 15.4156L19.6472 15.6793L20.9441 23.07L20.9441 23.0702C20.9716 23.2264 20.9099 23.3546 20.8149 23.4299C20.7219 23.5036 20.6094 23.5201 20.4993 23.4637C20.4993 23.4637 20.4992 23.4637 20.4991 23.4637L13.6385 19.9388L13.4099 19.8214L13.1814 19.9389L6.32555 23.4637L5.88061 23.07Z"
                fill="url(#paint3/img/image1.jpg3_linear_73_120)"
                stroke="white"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="paint3/img/image1.jpg3_linear_73_120"
                  x1="0.91211"
                  x2="25.5049"
                  y1="11.7199"
                  y2="11.7199"
                >
                  <stop
                    stop-color="white"
                  />
                  <stop
                    offset="1"
                    stop-color="white"
                  />
                  <stop
                    offset="1"
                    stop-color="white"
                    stop-opacity="0"
                  />
                </lineargradient>
              </defs>
            </svg>
            <svg
              class="Ratingbar__rait__star "
              fill="none"
              height="24"
              role="svg"
              viewBox="0 0 26 24"
              width="26"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.88061 23.07L5.88056 23.0702C5.85306 23.2264 5.91485 23.3546 6.00976 23.4299C6.1027 23.5036 6.21513 23.5201 6.32514 23.4639L5.88061 23.07ZM5.88061 23.07L7.17748 15.6793L7.22376 15.4156L7.02987 15.2309L1.52553 9.98747C1.52549 9.98743 1.52544 9.98739 1.5254 9.98734C1.41972 9.88639 1.38802 9.74107 1.42943 9.60208C1.47062 9.46381 1.56872 9.37728 1.69212 9.35993L1.69285 9.35982L9.34597 8.27232L9.60439 8.2356L9.72199 8.00259L13.1344 1.24179C13.2573 0.998504 13.5674 0.998504 13.6903 1.24179L17.1027 8.00259L17.2203 8.2356L17.4787 8.27232L25.1318 9.35982L25.1326 9.35993C25.256 9.37728 25.354 9.4638 25.395 9.60179C25.4362 9.74044 25.4045 9.88573 25.2982 9.98695L25.2979 9.98719L19.7948 15.2309L19.6009 15.4156L19.6472 15.6793L20.9441 23.07L20.9441 23.0702C20.9716 23.2264 20.9099 23.3546 20.8149 23.4299C20.7219 23.5036 20.6094 23.5201 20.4993 23.4637C20.4993 23.4637 20.4992 23.4637 20.4991 23.4637L13.6385 19.9388L13.4099 19.8214L13.1814 19.9389L6.32555 23.4637L5.88061 23.07Z"
                fill="url(#paint3/img/image1.jpg4_linear_73_120)"
                stroke="white"
              />
              <defs>
                <lineargradient
                  gradientUnits="userSpaceOnUse"
                  id="paint3/img/image1.jpg4_linear_73_120"
                  x1="0.91211"
                  x2="25.5049"
                  y1="11.7199"
                  y2="11.7199"
                >
                  <stop
                    stop-color="white"
                  />
                  <stop
                    offset="0.01"
                    stop-color="white"
                  />
                  <stop
                    offset="0.02"
                    stop-color="white"
                    stop-opacity="0"
                  />
                </lineargradient>
              </defs>
            </svg>
          </div>
          <div
            class="Ratingbar__text"
          >
            Рейтинг 
            5.00
             | 
            3
             
          </div>
        </div>
        <div
          class="Details__split__p"
        >
          <div
            class="Fav "
          >
            Добавить в избранное
            <svg
              class="Fav__star"
              fill="none"
              height="51"
              role="star"
              viewBox="0 0 50 51"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                clip-path="url(#clip0_2_7)"
              >
                <path
                  d="M11.2874 49.2117C10.0812 49.8305 8.71244 48.7461 8.95619 47.3617L11.5499 32.5805L0.540567 22.093C-0.487558 21.1117 0.0468174 19.318 1.42494 19.1242L16.7312 16.9492L23.5562 3.42737C24.1718 2.20862 25.8374 2.20862 26.4531 3.42737L33.2781 16.9492L48.5843 19.1242C49.9624 19.318 50.4968 21.1117 49.4656 22.093L38.4593 32.5805L41.0531 47.3617C41.2968 48.7461 39.9281 49.8305 38.7218 49.2117L24.9999 42.1617L11.2874 49.2117Z"
                  style="stroke: white;"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_2_7"
                >
                  <rect
                    fill="white"
                    height="50"
                    transform="translate(0 0.952362)"
                    width="50"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
        </div>
      </div>
      <img
        alt="image"
        class="Details__image"
        data-nimg="1"
        decoding="async"
        fetchpriority="high"
        height="0"
        sizes="100vw"
        src="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=3840&q=100"
        srcset="/_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=640&q=100 640w, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=750&q=100 750w, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=828&q=100 828w, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=1080&q=100 1080w, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=1200&q=100 1200w, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=1920&q=100 1920w, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=2048&q=100 2048w, /_next/image?url=%2Fimg%2F%2Fimg%2Fimage1.jpg&w=3840&q=100 3840w"
        style="color: transparent;"
        width="0"
      />
      <div
        class="Details__text"
      >
        sometext
      </div>
    </div>
  </section>
</div>
`;
